### PR TITLE
Use Github actions for CI

### PR DESCRIPTION
**Pull request by kmcphillips**
Since Travis is no longer really running CI for open source, switch this project to GH actions for CI.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        gemfile:
+          - Gemfile
+          - gemfiles/activesupport-5.0.gemfile
+          - gemfiles/activesupport-5.1.gemfile
+          - gemfiles/activesupport-6.0.gemfile
+    name: Ruby ${{ matrix.ruby }} ${{ matrix.gemfile }}
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Ruby ${{ matrix.ruby }}
+        uses: actions/setup-ruby@v1
```
Comment (chrisseaton) [inline .github/workflows/ci.yml:24]: Did you decide to use the old `setup-ruby` for a specific reason? This is the one we said wasn't well maintained.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - name: Set up gemfile
+        run: |
+          export BUNDLE_GEMFILE=${GITHUB_WORKSPACE}/${{ matrix.gemfile }}
```
Comment (peterzhu2118) [inline .github/workflows/ci.yml:29]: IIRC GitHub Actions does not persist `export`'d environment variables between steps. You can see this in the logs, for example logs for run "Ruby 2.5 gemfiles/activesupport-5.1.gemfile" shows `Installing activesupport 6.1.0`.

I think you'll need to use [`jobs.build.env`](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idenv).

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          rm -f ${GITHUB_WORKSPACE}/Gemfile.lock
```
Comment (peterzhu2118) [inline .github/workflows/ci.yml:30]: This is before `bundle install` is ran, is there a `Gemfile.lock` there?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (chrisseaton) [reply .github/workflows/ci.yml:24]: For example https://github.com/Shopify/quilt/pull/1693

Comment (kmcphillips) [reply .github/workflows/ci.yml:24]: No, just confusion over naming. I'll switch to `ruby/setup-ruby@v1`.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (kmcphillips) [reply .github/workflows/ci.yml:30]: No, it's not there now. And it's not necessary. I saw it in `shopify/shopify_api` and I like it as a defence against one getting committed accidentally. I can remove it.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kmcphillips) [reply .github/workflows/ci.yml:29]: Super helpful. Thanks.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          - gemfiles/activesupport-5.0.gemfile
+          - gemfiles/activesupport-5.1.gemfile
+          - gemfiles/activesupport-6.0.gemfile
+    name: Ruby ${{ matrix.ruby }} ${{ matrix.gemfile }}
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
```
Comment (paracycle) [inline .github/workflows/ci.yml:28]: I highly suggest you use `bundler-cache: true` to do automatic `bundle install` and cache the resulting bundle between runs: https://github.com/ruby/setup-ruby#bundler

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Run tests
+        run: |
+          gem install bundler
+          bundle install --jobs 4 --retry 3
+          bundle exec rake
```
Comment (paracycle) [inline .github/workflows/ci.yml:33]: If you use `bundler-cache: true` then this becomes:
```suggestion
          bundle exec rake
```

**Comment by kmcphillips** [comment]
I learned about 100 things about GH actions today. I think I'm happy with this config if someone wants to give a last 👀

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**